test/puppetlabs/puppetdb/integration/reports.clj

```diff
@@ -101,6 +101,86 @@
                       (int/pql-query pdb)
                       count))))))))
 
+(when tu/test-rich-data?
+  (def rich-data-tests
+    ; Names must be unique, they are used as Resource titles
+    [
+     ; for undef and false a wrapper is required or message is interpreted as 'absent' and no change occurs
+     {:name "literal_undef" :code "{ 'wrapper' => [undef] }" :expected "{\"wrapper\"=>[nil]}"}
+     {:name "literal_false" :code "{ 'wrapper' => false }" :expected "{\"wrapper\"=>false}"}
+
+     {:name "literal_default" :code "default" :expected "default"}
+     {:name "literal_integer" :code "47" :expected "47"}
+     {:name "literal_float" :code "3.14" :expected "3.14"}
+     {:name "literal_true" :code "true" :expected "true"}
+     {:name "literal_string" :code "\"hello\"" :expected "hello"}
+     {:name "string_with_single_quote" :code "\"ta'phoenix\"" :expected "ta'phoenix"}
+     {:name "string_with_double_quote" :code "\"he said \\\"hi\\\"\"" :expected "he said \"hi\""}
+     {:name "regexp" :code "/[a-z]+/" :expected "/[a-z]+/"}
+     {:name "deferred" :code "Deferred('join', [[1, 2, 3], ':'])" :expected "1:2:3"}
+     {:name "sensitive_deferred" :code "Sensitive(Deferred('join', [[1, 2, 3], ':']))" :expected-old-value "[redacted]" :expected "[redacted]"}
+     {:name "sensitive" :code "Sensitive('password')" :expected-old-value "[redacted]" :expected "[redacted]"}
+     {:name "timestamp" :code "Timestamp('2012-10-10')" :expected "2012-10-10T00:00:00.000000000 UTC"}
+     {:name "hash_and_array" :code "{'a' => [1, 2, 3], 'b' => 'hello'}" :expected "{\"a\"=>[1, 2, 3], \"b\"=>\"hello\"}"}
+     {:name "special_key" :code "{ ['special', 'key'] => 10 }" :expected "{\"[\\\"special\\\", \\\"key\\\"]\"=>10}"}
+     {:name "hash_with_sensitive_val" :code "{ x => Sensitive(hush) }" :expected #"\{\"x\"=>\"#<Sensitive \[value redacted\]:[0-9]+>\"}"}
+     {:name "hash_with_sensitive_key" :code "{Sensitive(hush) => 42 }" :expected #"\{\"#<Sensitive \[value redacted\]:[0-9]+>\"=>42\}"}
```

Comment:
Perhaps make these longer ones at least two lines so they're easily visible in gh, etc.